### ISSUE TITLE
Linking error: LNK1181: cannot open input file 'DirectXTex.lib

### ISSUE THREAD

**Issue by tincivilfx** [ash3D/VideoRecorder#1]
1>------ Rebuild All started: Project: DirectXTex, Configuration: Release x64 ------
1>DirectXTexUtil.cpp
1>BC.cpp
1>BC4BC5.cpp
1>BC6HBC7.cpp
1>BCDirectCompute.cpp
1>DirectXTexCompress.cpp
1>DirectXTexCompressGPU.cpp
1>DirectXTexConvert.cpp
1>DirectXTexD3D11.cpp
1>DirectXTexD3D12.cpp
1>DirectXTexDDS.cpp
1>DirectXTexFlipRotate.cpp
1>DirectXTexHDR.cpp
1>DirectXTexImage.cpp
1>DirectXTexMipMaps.cpp
1>DirectXTexMisc.cpp
1>DirectXTexNormalMaps.cpp
1>DirectXTexPMAlpha.cpp
1>DirectXTexResize.cpp
1>DirectXTexTGA.cpp
1>DirectXTexWIC.cpp
1>Generating Code...
1>DirectXTex_Desktop_2019_Win10.vcxproj -> ...\VideoRecorder\DirectXTex\DirectXTex\Bin\Desktop_2019_Win10\x64\Release\DirectXTex.lib
2>------ Rebuild All started: Project: VideoRecorder, Configuration: Release x64 ------
2>VideoRecorder.cpp
2>...\VideoRecorder\VideoRecorder.cpp(79,4): warning C4996: 'av_register_all': was declared deprecated
2>...\VideoRecorder\VideoRecorder.cpp(80,4): warning C4996: 'avcodec_register_all': was declared deprecated
2>...\VideoRecorder\VideoRecorder.cpp(468,44): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
2>...\VideoRecorder\VideoRecorder.cpp(468,44): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data
2>...\VideoRecorder\VideoRecorder.cpp(595,67): warning C4996: 'std::codecvt_utf8<wchar_t,1114111,(std::codecvt_mode)0>': warning STL4017: std::wbuffer_convert, std::wstring_convert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\codecvt(35,92): message : see declaration of 'std::codecvt_utf8'
2>...\VideoRecorder\VideoRecorder.cpp(595,46): warning C4996: 'std::wstring_convert<std::codecvt_utf8<wchar_t,1114111,(std::codecvt_mode)0>,wchar_t,std::allocator<wchar_t>,std::allocator<char>>': warning STL4017: std::wbuffer_convert, std::wstring_convert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xlocbuf(296,55): message : see declaration of 'std::wstring_convert'
2>...\VideoRecorder\VideoRecorder.cpp(595,91): warning C4996: 'std::wstring_convert<std::codecvt_utf8<wchar_t,1114111,(std::codecvt_mode)0>,wchar_t,std::allocator<wchar_t>,std::allocator<char>>::to_bytes': warning STL4017: std::wbuffer_convert, std::wstring_convert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.
2>LINK : fatal error LNK1181: cannot open input file 'DirectXTex.lib'
2>Done building project "VideoRecorder.vcxproj" -- FAILED.
========== Rebuild All: 1 succeeded, 1 failed, 0 skipped ==========

**Comment by tincivilfx** [comment]
Need to fix linker path reference:
![Capture](https://user-images.githubusercontent.com/43148009/81351477-4584e000-9079-11ea-9784-8f2a27e0d532.png)

**Comment by sahilsharma-dot** [comment]
fatal error LNK1181: cannot open input file 'avcodec.lib'

**Comment by sahilsharma-dot** [comment]
their is no avcodec.lib in C:/ffmpeg/
their is exist avcode.h
does it requires to create avcodec.lib

**Comment by ash3D** [comment]
@sahilsharma-dot 
'avcodec.lib' should be in 'lib' directory, appropriate variant of ffmpeg build with development files contains it.
Also do not forget to specify path to ffmpeg in 'FFMPEG_ROOT' macro in property pages.

**Comment by sahilsharma-dot** [comment]
yes it was built the library on the same day. i forgot to. Mark the issue
solved sorry for that
Mr ash3d will it be able to record ui in android like when I project video
using android camera onto ui. Will it be able to record it
. I have to make libraries and lot of other stuff for android and I think
if it is possible than I will definately go for
thanks for replying quickly

**Comment by ash3D** [comment]
@sahilsharma-dot 
Do you use this lib diretly or build it for [Unreal Engine plugin](https://github.com/ash3D/UEVideoRecorder)?
This lib just takes bitmap and encodes it to video stream. It's completely up to you for what pixels you are feeding if you use it directly.
If you are interested in capturing Unreal Engine's UI, [UEVideoRecorder](https://github.com/ash3D/UEVideoRecorder) has trick to support UI capturing if 'CaptureGUI' setting enabled. It worked well with older UE4 version on Windows DX11. I didn't tested it with recent Unreal Engine versions and/or other UI subsystems. If they behave differenly other handling can potentially be required for UI.

**Comment by sahilsharma-dot** [comment]
thnaks. I will post here if it works.
